core/observables/observable.py

```diff
@@ -128,7 +128,12 @@ def add_text(cls, text, tags=[]):
             A saved Observable instance.
 
         """
-        o = Observable.guess_type(text).get_or_create(value=text)
+        if type is None:
+            guessed_type = Observable.guess_type(text)
+        else:
+            guessed_type = instantiate_subclass(Observable, type)
+
+        o = guessed_type.get_or_create(value=text)
```

Comment:
If `type` is provided, we're not guessing anymore, so I'd change `guessed_type` to `observable_type` or something more generic that does not denote guessing.

Reply:
I followed your recommandation